### PR TITLE
test(migrate): tweak setup db scripts

### PR DESCRIPTION
**Pull request by Jolg42**
To avoid this in our test suite (somehow only happening on macOS)
https://github.com/prisma/prisma/runs/5009329376?check_suite_focus=true#step:13:761
```
   ● postgresql › basic introspection

    expect(received).toMatchInlineSnapshot(snapshot)

    Snapshot name: `postgresql basic introspection 3`

    - Snapshot  - 0
    + Received  + 1

    + Error: P2002: Unique constraint failed on the fields: (`datname`)

```

### CODE REVIEW

**Comment by Jolg42** [comment]
The flaky windows snapshots should be caused by the fact that the engine implementation sends the result from the RPC request via logs and not in the response, we will change that after the release and can be ignored here.

File: packages/migrate/src/__tests__/MigrateDev.test.ts [hunk 1]
```diff
@@ -891,28 +891,22 @@ describe('sqlite', () => {
...
+  const connectionString = (
+    process.env.TEST_POSTGRES_URI_MIGRATE || 'postgres://prisma:prisma@localhost:5432/tests-migrate'
+  ).replace('tests-migrate', 'tests-migrate-dev')
+
+  // Update env var because it's the one that is used in the schemas tested
+  process.env.TEST_POSTGRES_URI_MIGRATE = connectionString
+  process.env.TEST_POSTGRES_SHADOWDB_URI_MIGRATE = connectionString.replace(
+    'tests-migrate-dev',
+    'tests-migrate-dev-shadowdb',
+  )
```
Comment (Jolg42) [inline packages/migrate/src/__tests__/MigrateDev.test.ts:903]: I'm not happy with this, I think we can improve things here but later

File: packages/migrate/src/utils/setupMysql.ts [hunk 2]
```diff
@@ -13,27 +13,33 @@ export async function setupMysql(options: SetupParams): Promise<void> {
...
     host: credentials.host,
     port: credentials.port,
-    database: credentials.database,
+    // database: credentials.database, // use the default db
     user: credentials.user,
     password: credentials.password,
     multipleStatements: true,
   })
+  await dbDefault.query(`
+CREATE DATABASE IF NOT EXISTS \`${credentials.database}-shadowdb\`;
```
Comment (millsp) [inline packages/migrate/src/utils/setupMysql.ts:26]: No need to drop previous/existing dbs?

File: packages/migrate/src/__tests__/MigrateDev.test.ts [hunk 3]
```diff
@@ -1126,17 +1120,22 @@ describe('postgresql', () => {
 })
 
 describe('mysql', () => {
+  const connectionString = (
+    process.env.TEST_MYSQL_URI_MIGRATE || 'mysql://root:root@localhost:3306/tests-migrate'
+  ).replace('tests-migrate', 'tests-migrate-dev')
+
+  // Update env var because it's the one that is used in the schemas tested
+  process.env.TEST_MYSQL_URI_MIGRATE = connectionString
```
Comment (millsp) [inline packages/migrate/src/__tests__/MigrateDev.test.ts:1128]: I prefer this format https://github.com/prisma/prisma/pull/11524/files#diff-c266a026f47d4de125b7139076b575b33fb6dcb0207110c4a9a0f78304097b41R1335

**Review by aqrln — APPROVED** [review]

File: packages/migrate/src/utils/setupPostgres.ts [hunk 4]
```diff
@@ -13,23 +13,26 @@ export async function setupPostgres(options: SetupParams): Promise<void> {
...
-  await createDatabase(connectionString).catch((e) => console.error(e))
-
-  const db = new Client({
-    connectionString: connectionString,
+  // Connect to default db
+  const dbDefault = new Client({
+    connectionString: connectionString.replace(credentials.database!, 'postgres'),
   })
+  await dbDefault.connect()
+  await dbDefault.query(`DROP DATABASE IF EXISTS "${credentials.database}-shadowdb";`)
```
Comment (millsp) [inline packages/migrate/src/utils/setupPostgres.ts:21]: Why does PG need to be dropped, but not other DBs?

**Review by millsp — APPROVED** [review]

File: packages/migrate/src/__tests__/MigrateDev.test.ts (hunk 3 above)
Comment (aqrln) [reply packages/migrate/src/__tests__/MigrateDev.test.ts:1128]: Yeah, I have no super strong opinion on this, but I personally think the default hardcoded connection strings are not super useful and I'd personally prefer the test to fail if the env var is not set. That the absence of the default value would make the second format (which is shorter and more readable) possible here is a nice bonus.

Comment (aqrln) [reply packages/migrate/src/__tests__/MigrateDev.test.ts:1128]: Although that format is possible even with the default value, like this:

```ts
process.env.TEST_MYSQL_URI_MIGRATE = process.env.TEST_MYSQL_URI_MIGRATE?.replace(
  'tests-migrate',
  'tests-migrate-dev'
) ?? 'mysql://root:root@localhost:3306/tests-migrate-dev'
```

Comment (Jolg42) [reply packages/migrate/src/__tests__/MigrateDev.test.ts:1128]: Yes I think we can get rid of all the defaults at some point in a separate PR @aqrln, it's hard to maintain them (might not be in sync) and a clear single point of config is better

File: packages/migrate/src/utils/setupPostgres.ts (hunk 4 above)
Comment (Jolg42) [reply packages/migrate/src/utils/setupPostgres.ts:21]: It's my way of doing a create table if not exists which doesn't work on PostgreSQL 😢 
So it always starts clean

**Pull request merged**